backend/app/controllers/spree/admin/menu_items_controller.rb

```diff
@@ -0,0 +1,25 @@
+module Spree
+  module Admin
+    class MenuItemsController < ResourceController
+      belongs_to 'spree/menu'
+
+      before_action :load_data
+
+      def collection_url
+        spree.edit_admin_menu_path(@menu)
+      end
+
+      def location_after_save
+        spree.edit_admin_menu_menu_item_path(@menu, @menu_item)
+      end
+
+      def load_data
+        @menu_item_types = Spree::MenuItem::ITEM_TYPE
+        @linked_resource_types = Spree::MenuItem::LINKED_RESOURCE_TYPE
+
+        @taxons = Spree::Taxon.all
+        @products = Spree::Product.all
```

Comment:
@MatthewKennedy we need to rather use Platform API endpoints to fetch Products and Taxons, loading so many records into the memory will slow things down